apps/namadillo/src/App/Transactions/TransactionHistory.tsx

```diff
@@ -60,25 +85,37 @@ export const TransactionHistory = (): JSX.Element => {
       handleFiltering(transaction)
     ) ?? [];
 
+  const allHistoricalTransactions = [
+    ...historicalTransactions,
+    ...completedIbcShieldTransactions,
+  ].sort((a, b) => (b?.timestamp ?? 0) - (a?.timestamp ?? 0));
+
   // Calculate total pages based on the filtered transactions
   const totalPages = Math.max(
     1,
-    Math.ceil(historicalTransactions.length / ITEMS_PER_PAGE)
+    Math.ceil(allHistoricalTransactions.length / ITEMS_PER_PAGE)
   );
 
   const paginatedTransactions = useMemo(() => {
     const startIndex = currentPage * ITEMS_PER_PAGE;
     const endIndex = startIndex + ITEMS_PER_PAGE;
-    return historicalTransactions.slice(startIndex, endIndex);
-  }, [historicalTransactions, currentPage]);
+    return allHistoricalTransactions.slice(startIndex, endIndex);
+  }, [allHistoricalTransactions, currentPage]);
 
   const renderRow = (
     transaction: TransactionHistoryType,
     index: number
   ): TableRow => {
     return {
       key: transaction.tx?.txId || index.toString(),
-      cells: [<TransactionCard key="transaction" tx={transaction} />],
+      cells: [
+        transaction?.tx ?
+          <TransactionCard key="transaction" tx={transaction} />
+        : <LocalStorageTransactionCard
+            key="transaction"
+            transaction={transaction as unknown as TransferTransactionData}
```

Comment:
Not a great fan but has to be done since we're mixing two completely different types of transactions (local storage & RPC)